release flux versions 0.4.3-0.8.2

saada opened the pull request:
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [x] Does your operator have [nested directory structure](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#create-a-bundle)?
* [ ] Have you selected the Project *Community Operator Submissions* in your PR on the right-hand menu bar?
* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md)?
* [x] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?
* [x] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#operator-icon) set?

### Updates to existing Operators

* [x] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [x] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [x] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#bundle-format) defined in the `package.yaml` or `annotations.yaml` ?
* [x] Have you tested an update to your Operator when deployed via OLM?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?

### Your submission should not

* [x] Modify more than one operator
* [x] Modify an Operator you don't own
* [x] Rename an operator - please remove and add with a different name instead
* [x] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-required-fields.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-capability-level.svg)

saada commented:
The output from "Operator test / kiwi / Full operator test" is not clear. Can we get some feedback to help understand the error?

mvalarh commented:
@J0zi can you help please?

J0zi commented:
@saada Standar procedure is to bump an operator version in case of major changes. I don't want to say, you cannot do this, but you are overwriting all existing version, some already installed on customer sides. It can be a case, where customer has a working operator, then you overwrite it and if it does not work, customer cannot install an older version, because all are changed.
We randomly tested `0.6.1` and it is failing. Maybe because it requires far more RAM than the last release (0.4.2). If this is the case for all, than requirements has changed, it can be an issue for customer. But you should decide.

We had also problem to start 0.7.7 in default time on default machine. What your own tests say?

mvalarh commented:
@saada if you are ready for review. Please press "Ready for review". Then tests are going to be run again

saada commented:
I'm trying to test locally to understand the failing test issue but CRC has been nearly impossible to spin up

J0zi commented:
@saada, please check manual testing here https://github.com/operator-framework/community-operators/blob/master/docs/operator-test-suite.md

saada commented:
Tried running tests locally and got the following error. Also, is it trying to edit my /etc/hosts?

```
TASK [reset_kind : Configuring Kind with registry] *****************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/kind-with-registry.sh", "delta": "0:00:04.770908", "end": "2021-02-16 07:41:54.130696", "msg": "non-zero return code", "rc": 1, "start": "2021-02-16 07:41:49.359788", "stderr": "Error: No such container: kind-registry\nUnable to find image 'registry:2' locally\n2: Pulling from library/registry\n0a6724ff3fcd: Pulling fs layer\nd550a247d74f: Pulling fs layer\n1a938458ca36: Pulling fs layer\nacd758c36fc9: Pulling fs layer\n9af6d68b484a: Pulling fs layer\nacd758c36fc9: Waiting\n9af6d68b484a: Waiting\nd550a247d74f: Verifying Checksum\nd550a247d74f: Download complete\n0a6724ff3fcd: Verifying Checksum\n0a6724ff3fcd: Download complete\nacd758c36fc9: Verifying Checksum\nacd758c36fc9: Download complete\n0a6724ff3fcd: Pull complete\n9af6d68b484a: Verifying Checksum\n9af6d68b484a: Download complete\nd550a247d74f: Pull complete\n1a938458ca36: Verifying Checksum\n1a938458ca36: Download complete\n1a938458ca36: Pull complete\nacd758c36fc9: Pull complete\n9af6d68b484a: Pull complete\nDigest: sha256:d5459fcb27aecc752520df4b492b08358a1912fcdfa454f7d2101d4b09991daa\nStatus: Downloaded newer image for registry:2\nsed: 1: \"/etc/hosts\": extra characters at the end of h command", "stderr_lines": ["Error: No such container: kind-registry", "Unable to find image 'registry:2' locally", "2: Pulling from library/registry", "0a6724ff3fcd: Pulling fs layer", "d550a247d74f: Pulling fs layer", "1a938458ca36: Pulling fs layer", "acd758c36fc9: Pulling fs layer", "9af6d68b484a: Pulling fs layer", "acd758c36fc9: Waiting", "9af6d68b484a: Waiting", "d550a247d74f: Verifying Checksum", "d550a247d74f: Download complete", "0a6724ff3fcd: Verifying Checksum", "0a6724ff3fcd: Download complete", "acd758c36fc9: Verifying Checksum", "acd758c36fc9: Download complete", "0a6724ff3fcd: Pull complete", "9af6d68b484a: Verifying Checksum", "9af6d68b484a: Download complete", "d550a247d74f: Pull complete", "1a938458ca36: Verifying Checksum", "1a938458ca36: Download complete", "1a938458ca36: Pull complete", "acd758c36fc9: Pull complete", "9af6d68b484a: Pull complete", "Digest: sha256:d5459fcb27aecc752520df4b492b08358a1912fcdfa454f7d2101d4b09991daa", "Status: Downloaded newer image for registry:2", "sed: 1: \"/etc/hosts\": extra characters at the end of h command"], "stdout": "9dc574ddeb8f98252b5910e2be5240122a1bc9752ebcbf61ef81a42882284932", "stdout_lines": ["9dc574ddeb8f98252b5910e2be5240122a1bc9752ebcbf61ef81a42882284932"]}

TASK [reset_kind : fail] *******************************************************
skipping: [localhost]

TASK [reset_kind : debug] ******************************************************
ok: [localhost] => {
    "msg": "Failed to install kind and registry - Retrying (3/5) ..."
}

TASK [reset_kind : include_tasks] **********************************************
included: /Users/saada/.ansible/pull/1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa/upstream/roles/reset_kind/tasks/reset.yml for localhost

TASK [reset_kind : Set the retry count] ****************************************
ok: [localhost]

TASK [reset_kind : Check that the kind executable exists] **********************
ok: [localhost]

TASK [reset_kind : Delete kind cluster] ****************************************
changed: [localhost]

TASK [reset_kind : Destroy current kind registry] ******************************
changed: [localhost]

TASK [reset_kind : Generating file for kind and registry] **********************
ok: [localhost]

TASK [reset_kind : Configuring Kind with registry] *****************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/kind-with-registry.sh", "delta": "0:00:00.741256", "end": "2021-02-16 07:41:58.685964", "msg": "non-zero return code", "rc": 1, "start": "2021-02-16 07:41:57.944708", "stderr": "Error: No such container: kind-registry\nsed: 1: \"/etc/hosts\": extra characters at the end of h command", "stderr_lines": ["Error: No such container: kind-registry", "sed: 1: \"/etc/hosts\": extra characters at the end of h command"], "stdout": "b2d9e985c25c658fcdf791a82259ac2711778c120d954075a9e2475187ae6542", "stdout_lines": ["b2d9e985c25c658fcdf791a82259ac2711778c120d954075a9e2475187ae6542"]}

TASK [reset_kind : fail] *******************************************************
skipping: [localhost]

TASK [reset_kind : debug] ******************************************************
ok: [localhost] => {
    "msg": "Failed to install kind and registry - Retrying (4/5) ..."
}

TASK [reset_kind : include_tasks] **********************************************
included: /Users/saada/.ansible/pull/1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa/upstream/roles/reset_kind/tasks/reset.yml for localhost

TASK [reset_kind : Set the retry count] ****************************************
ok: [localhost]

TASK [reset_kind : Check that the kind executable exists] **********************
ok: [localhost]

TASK [reset_kind : Delete kind cluster] ****************************************
changed: [localhost]

TASK [reset_kind : Destroy current kind registry] ******************************
changed: [localhost]

TASK [reset_kind : Generating file for kind and registry] **********************
ok: [localhost]

TASK [reset_kind : Configuring Kind with registry] *****************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/kind-with-registry.sh", "delta": "0:00:00.681224", "end": "2021-02-16 07:42:02.804550", "msg": "non-zero return code", "rc": 1, "start": "2021-02-16 07:42:02.123326", "stderr": "Error: No such container: kind-registry\nsed: 1: \"/etc/hosts\": extra characters at the end of h command", "stderr_lines": ["Error: No such container: kind-registry", "sed: 1: \"/etc/hosts\": extra characters at the end of h command"], "stdout": "d3b7397148a1f25efd21b5641d374aadb9ec6969b9665d80f8478e8124eb87bc", "stdout_lines": ["d3b7397148a1f25efd21b5641d374aadb9ec6969b9665d80f8478e8124eb87bc"]}

TASK [reset_kind : fail] *******************************************************
skipping: [localhost]

TASK [reset_kind : debug] ******************************************************
ok: [localhost] => {
    "msg": "Failed to install kind and registry - Retrying (5/5) ..."
}

TASK [reset_kind : include_tasks] **********************************************
included: /Users/saada/.ansible/pull/1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa/upstream/roles/reset_kind/tasks/reset.yml for localhost

TASK [reset_kind : Set the retry count] ****************************************
ok: [localhost]

TASK [reset_kind : Check that the kind executable exists] **********************
ok: [localhost]

TASK [reset_kind : Delete kind cluster] ****************************************
changed: [localhost]

TASK [reset_kind : Destroy current kind registry] ******************************
changed: [localhost]

TASK [reset_kind : Generating file for kind and registry] **********************
ok: [localhost]

TASK [reset_kind : Configuring Kind with registry] *****************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/kind-with-registry.sh", "delta": "0:00:00.710827", "end": "2021-02-16 07:42:07.111332", "msg": "non-zero return code", "rc": 1, "start": "2021-02-16 07:42:06.400505", "stderr": "Error: No such container: kind-registry\nsed: 1: \"/etc/hosts\": extra characters at the end of h command", "stderr_lines": ["Error: No such container: kind-registry", "sed: 1: \"/etc/hosts\": extra characters at the end of h command"], "stdout": "73172d62d8f65e65115201d546fe08091382db89e0066316c667ee006def494d", "stdout_lines": ["73172d62d8f65e65115201d546fe08091382db89e0066316c667ee006def494d"]}

TASK [reset_kind : fail] *******************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Ended after 5 retries"}

PLAY RECAP *********************************************************************
localhost                  : ok=38   changed=11   unreachable=0    failed=1    skipped=8    rescued=5    ignored=0   
```

mvalarh commented:
@saada I would test it on your repo (because it is not in our master yet)
```
bash <(curl -sL https://cutt.ly/WhkV76k) \
kiwi \
upstream-community-operators/flux/v0.7.7 \
https://github.com/saada/community-operators flux-releases-0.7.7
```
But i see you have problem to start kind cluster + registry. What OS are you using? Can you provide output when you run script `/tmp/kind-with-registry.sh`?

saada commented:
I get
```sh
sed: 1: "/etc/hosts": extra characters at the end of h command
```

saada commented:
The issue seems to be that the operator-sdk binary that gets installed is not compatible with macOS?

```sh
TASK [reset_kind : Installing OLM (0.17.0)] ************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/operator-test/bin/operator-sdk olm install --version 0.17.0 --timeout 5m0s", "delta": "0:00:00.006577", "end": "2021-02-18 07:35:37.428087", "msg": "non-zero return code", "rc": 126, "start": "2021-02-18 07:35:37.421510", "stderr": "/bin/sh: /tmp/operator-test/bin/operator-sdk: cannot execute binary file", "stderr_lines": ["/bin/sh: /tmp/operator-test/bin/operator-sdk: cannot execute binary file"], "stdout": "", "stdout_lines": []}
```

saada commented:
Got the kiwi tests to pass locally. I also added the latest releases since this PR started for 0.8.0 and 0.8.1.

J0zi commented:
@saada - to move upstream-community-operators` to `community-operators`, just open a separate PR with the operator in `community-operators` directory.

saada commented:
Thanks. I think this is ready to merge 🚀

J0zi commented:
@saada - did you already tested locally the newest version `0.8.2`? Our pipeline was not able find the latest operator for testing, 500 changed files is somehow beyond limits. Were you lucky to have the latest version green locally?

J0zi commented:
I tested it locally and kiwi is green, orange is not needed as we are rebuilding everything from scratch, also lemon is green.

saada commented:
Yup I confirmed it works locally as well. Thank you so much.
Here's the same PR against community-operators https://github.com/operator-framework/community-operators/pull/3168
